docs/DafnyRef/Types.md

```diff
@@ -870,18 +870,18 @@ of type `T`, integer-based numeric `i` satisfying `0 <= i < |s|`, and
 integer-based numerics `lo` and `hi` satisfying
 `0 <= lo <= hi <= |s|`, sequences support the following operations:
 
- expression         | result type | description
- ---------------------|:---:|----------------------------------------
- `|s|`               | `nat` | sequence length
- `s[i]`              | `T` |sequence selection
- `s[i := e]`         | `seq<T>` | sequence update
- `e in s`            | `bool` | sequence membership
- `e !in s`           | `bool` | sequence non-membership
- `s[lo..hi]`         | `seq<T>`| subsequence
- `s[lo..]`           | `seq<T>` | drop
- `s[..hi]`           | `seq<T>` | take
- `s[`_slices_`]`   | `seq<seq<T>>` | slice
- `multiset(s)`       | `multiset<T>`| sequence conversion to a `multiset<T>`
+ expression         | precedence | result type | description
+ -------------------|:---:|:---:|----------------------------------------
+ `e in s`           | 4 | `bool` | sequence membership
```

Comment:
Did you get these numbers from Parser.cs ? If they, they are subject to change as it's generated code.

Reply:
The numbers come from the reference manual itself. They are implicit in the tables of operations, but GitHub markdown does not render the horizontal rules that make the separation between operators clear. So I added a precedence number that is consistent across all the tables . It is possible that these could change with new operators being added, but they would have to have a whole new precedence group. So it would be rare. The Dafny parser builds precedence hard-coded into the grammar productions.